### PR TITLE
[webnn] Add float32 tests for WebNN API layerNormalization op

### PR DESCRIPTION
**Pull request by BruceDai**
@fdwr Please help provide this tolerance definition for **layerNormalization** op. 
And running test with **negative** value of 1D input by WebNN GPU backend (DirectML)  it would be **NaN** which is expected as **0**, synced with@shiyi9801, she said that you knew there's a bug for it on DirectML. Another problem, both **scale** and **bias** are required by DirectML side.

### CODE REVIEW

File: webnn/resources/utils.js
Comment (BruceDai) [inline webnn/resources/utils.js:325]: This is a placeholder.

**Comment by fdwr** [comment]
> both scale and bias are required by DirectML side.

They are? Both scale and bias are optional tensors in the API 🤔:

```
struct DML_MEAN_VARIANCE_NORMALIZATION1_OPERATOR_DESC
{
    const DML_TENSOR_DESC* InputTensor;
    _Maybenull_ const DML_TENSOR_DESC* ScaleTensor;
    _Maybenull_ const DML_TENSOR_DESC* BiasTensor;
    const DML_TENSOR_DESC* OutputTensor;
    UINT AxisCount;
    _Field_size_(AxisCount) const UINT* Axes;
    BOOL NormalizeVariance;
    FLOAT Epsilon;
    _Maybenull_ const DML_OPERATOR_DESC* FusedActivation;
};
```

> 1D input by WebNN GPU backend (DirectML) it would be NaN which is expected as 0, synced with@shiyi9801, she said that you knew there's a bug for it on DirectML.

I can also repro it with DxDispatch, but I haven't concluded whether it should be considered a *bug* or not, given we're essentially dividing 0 by 0 in `(Scale * ((Input - Mean) / sqrt(Variance + Epsilon)) + Bias`. ⏳

**Comment by BruceDai** [comment]
> They are? Both scale and bias are optional tensors in the API 🤔:

There's a below error: 
```
 "NotSupportedError: DirectML: layerNormalization: The scale and bias must be both given or not given."
```

@shiyi8901 Please also help a look, thanks.

File: webnn/resources/utils.js
Comment (fdwr) [inline webnn/resources/utils.js:667]: > There's a below error:  "NotSupportedError: DirectML: layerNormalization: The scale and bias must be both given or not given."

Ah, scale and bias are not independently optional, as you must provide *both* of them or *neither* of them to DML FL 4.0 (https://learn.microsoft.com/en-us/windows/win32/api/directml/ns-directml-dml_mean_variance_normalization_operator_desc "ScaleTensor - If DML_FEATURE_LEVEL is less than DML_FEATURE_LEVEL_5_2, then this tensor is required if BiasTensor is present."). This was relaxed in DML feature level 5.2+, but our target DML FL 4.0 still has this restriction.

🤔 So maybe the Chromium WebNN backend needs to create a constant tensor full of zeros under the hood if bias is not given. @huningxin, what do you think about that?

Comment (huningxin) [reply webnn/resources/utils.js:667]: > 🤔 So maybe the Chromium WebNN backend needs to create a constant tensor full of zeros under the hood if bias is not given. @huningxin, what do you think about that?

It sounds good to me. Thanks @fdwr !

WPT test cases should follow the spec that treat scale and bias are independently optional. We can file an issue against the current Chromium implementation for the NotSupportedError due to DML FL 4.0 restriction.

Comment (BruceDai) [reply webnn/resources/utils.js:667]: Opened Chromium issue https://bugs.chromium.org/p/chromium/issues/detail?id=1513041, please take a look, thanks.

**Comment by BruceDai** [comment]
> > both scale and bias are required by DirectML side.
> 
> They are? Both scale and bias are optional tensors in the API 🤔:
> 
> ```
> struct DML_MEAN_VARIANCE_NORMALIZATION1_OPERATOR_DESC
> {
>     const DML_TENSOR_DESC* InputTensor;
>     _Maybenull_ const DML_TENSOR_DESC* ScaleTensor;
>     _Maybenull_ const DML_TENSOR_DESC* BiasTensor;
>     const DML_TENSOR_DESC* OutputTensor;
>     UINT AxisCount;
>     _Field_size_(AxisCount) const UINT* Axes;
>     BOOL NormalizeVariance;
>     FLOAT Epsilon;
>     _Maybenull_ const DML_OPERATOR_DESC* FusedActivation;
> };
> ```
> 
> > 1D input by WebNN GPU backend (DirectML) it would be NaN which is expected as 0, synced with@shiyi9801, she said that you knew there's a bug for it on DirectML.
> 
> I can also repro it with DxDispatch, but I haven't concluded whether it should be considered a _bug_ or not, given we're essentially dividing 0 by 0 in `(Scale * ((Input - Mean) / sqrt(Variance + Epsilon)) + Bias`. ⏳

Soft ping @fdwr, any suggestion? Or do we firstly remove this test to not block this pr being merged? Thanks.

**Comment by fdwr** [comment]
> Soft ping @fdwr, any suggestion? Or do we firstly remove this test to not block this pr being merged? Thanks.

@BruceDai Please remove to unblock yourself. TY.

**Comment by BruceDai** [comment]
Thanks @fdwr. I just created an issue https://github.com/web-platform-tests/wpt/issues/43893 for this removing 1D test.

**Review by fdwr — APPROVED** [review]
Thank you for not force pushing! :) It makes reviewing the delta from the previous iteration so easy.

**Review by Honry — APPROVED** [review]
Thanks, LGTM!

**Pull request merged**
